### PR TITLE
Upgrade to Laravel 11.x

### PR DESCRIPTION
**Pull request by joelbutcher**
This PR allows Cachet to be installed in Laravel 11 apps.

### CODE REVIEW

**Comment by joelbutcher** [comment]
@jbrooksuk Not sure if your desire here is to support both L10 & L11, or if you want to just straight up drop L10.

I'm not sure what your release timelines are looking likes, but as you know, L10 no longer receives bug fixes and will stop receiving security fixes in Feb next year, so if you're not thinking of releasing until after then, it doesn't make much sense to me to keep supporting L10

**Comment by joelbutcher** [comment]
Also not sure why the migrations are only failing on L11 as they're the same migrations for cachet core as L10 🤔

**Comment by jbrooksuk** [comment]
I think tests fail due to changes in the way columns are changed? Unsure; but that's my gut feeling.

**Comment by joelbutcher** [comment]
@jbrooksuk fixed the migrations, I've also dropped L10 support and now only targeting L11... Can revert if you want, but personally think this is rather sensible due to risk of lack of support at time of release for Cachet V3

**Comment by joelbutcher** [comment]
Squashed the branch down to a single commit, now tests are failing because the `DatabaseSeeder` is being run for tests, which wasn't happening in L10. Anyone got any ideas?

**Comment by jbrooksuk** [comment]
Happy to support only Laravel 11.x, I think... but the cachet 3.x branch needs updating too. There is a PR that @gonedark made.

File: tests/TestCase.php [hunk 1]
```diff
@@ -31,4 +33,13 @@ protected function defineEnvironment($app)
...
     }
+
+    public static function cachedConfigurationForWorkbench(): Config
+    {
+        $config = Workbench::configuration();
+
+        $config['seeders'] = false;
+
+        return $config;
+    }
```
Comment (crynobone) [inline tests/TestCase.php:44]: Can you explain why you need to add this.

File: composer.json [hunk 2]
```diff
@@ -19,26 +19,26 @@
...
         "spatie/laravel-query-builder": "^5.5",
         "spatie/laravel-settings": "^3.2",
         "timacdonald/json-api": "^v1.0.0-beta.4",
         "twig/twig": "^3.0"
     },
     "require-dev": {
-        "orchestra/testbench": "^8.14",
-        "pestphp/pest": "^2.13",
-        "pestphp/pest-plugin-laravel": "^2.2",
+        "orchestra/testbench": "^9.5",
```
Comment (crynobone) [inline composer.json:39]:
```suggestion
        "orchestra/testbench": "^9.5.1",
```

File: tests/TestCase.php (hunk 1 above)
Comment (joelbutcher) [reply tests/TestCase.php:44]: Test seeders configured in the workbench config were running in the test suite. I think as a result of the `TestCase` class using the `WithWorkbench` trait. 

I've not done a lot of digging, but this fixed the issue. If the issue is deeper (or even a bug in testbench-core).

Comment (crynobone) [reply tests/TestCase.php:44]: `TestCase` wouldn't run `seeders` unless the `$seed` property is set to `true`. I can't find anywhere this is set for this repo.

https://github.com/laravel/framework/blob/2747a5ffe0d4dcdcc39b0cd3b3dd9892b619c0c4/src/Illuminate/Foundation/Testing/Traits/CanConfigureMigrationCommands.php#L50-L53

Comment (crynobone) [reply tests/TestCase.php:44]: Also, https://github.com/cachethq/core/pull/96/commits/baf12e52fccfd909d2856961a0be51001bb36288 would limit seeders to only be executed via `workbench:build` command.

Comment (joelbutcher) [reply tests/TestCase.php:44]: @crynobone i don't know what you're looking for from me. Try to pull this down and run the test suite with this removed. The seeders run and the tests that assert database counts exist fail 🤷

Comment (crynobone) [reply tests/TestCase.php:44]: Sorry, wanting to understand the issue as I maintain Testbench and curious to see what we can improves

Comment (joelbutcher) [reply tests/TestCase.php:44]: I appreciate that, thank you!

I did try to do some digging and found that the problem exists in the configuration returned from `Workbench::configuration()` [here](https://github.com/orchestral/testbench-core/blob/60dd2ad451cfcf3e85fcf15ff43d920b0f263ed3/src/Concerns/WithWorkbench.php#L27)

**Comment by joelbutcher** [comment]
@crynobone my local `main` was out of date. I've updated my fork and this branch is now good to go :)

**Review by crynobone — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -24,21 +24,21 @@
...
-        "illuminate/database": "^10.0",
-        "illuminate/events": "^10.0",
-        "illuminate/queue": "^10.0",
-        "illuminate/support": "^10.0",
+        "illuminate/console": "^11.0",
+        "illuminate/database": "^11.0",
+        "illuminate/events": "^11.0",
+        "illuminate/queue": "^11.0",
+        "illuminate/support": "^11.0",
         "nesbot/carbon": "^2.70",
```
Comment (crynobone) [inline composer.json:32]: Laravel 11 also supports Carbon 3 but I don't know if you want to cover that right now. 

Some additional breaking changes might make it worth a separate PR.

Comment (joelbutcher) [reply composer.json:32]: @crynobone thanks! Yeh I'll send in a follow up to isolate any BC problems there. 👍

**Comment by jbrooksuk** [comment]
@joelbutcher thanks for this! We need to get https://github.com/cachethq/cachet/pull/4384 reviewed and merged first.
